app/src/main/java/protect/card_locker/ScanActivity.java

```diff
@@ -190,19 +246,21 @@ public boolean onOptionsItemSelected(MenuItem item) {
         return super.onOptionsItemSelected(item);
     }
 
-    private void handleActivityResult(int requestCode, int resultCode, Intent intent) {
-        super.onActivityResult(requestCode, resultCode, intent);
-
-        BarcodeValues barcodeValues = Utils.parseSetBarcodeActivityResult(requestCode, resultCode, intent, this);
-
-        if (barcodeValues.isEmpty()) {
-            return;
+    private void setScannerActive(boolean isActive) {
+        if (isActive) {
+            barcodeScannerView.resume();
+            mScannerActive = true;
+        } else {
+            barcodeScannerView.pause();
+            mScannerActive = false;
         }
```

Comment:
Feels slightly cleaner. But I don't feel very strongly about it.

```suggestion
        if (isActive) {
            barcodeScannerView.resume();
        } else {
            barcodeScannerView.pause();
        }
        mScannerActive = isActive;
```